code/parachain/frame/composable-traits/src/staking/mod.rs

```diff
@@ -71,16 +104,20 @@ impl RewardRatePeriod {
 }
 
 /// A reward update states the new reward and reward_rate for a given asset
+// REVIEW(benluelo): Make this an enum with variants per reward type?
```

Comment:
Sounds like it would increase the complexity too much.

Reply:
The current model only allows us to update a reward to rate based, if we define an enum (even with only one variant for now) it would allow us to reuse the same extrinsic for updating a reward to any type

Reply:
I see, we don't really know the requirements for the future reward types to know if they need updates, so I think we can keep things as is for now :)